### PR TITLE
*: remove the support of the `amending transaction`

### LINKED ISSUE #40381: remove amend transaction
## Enhancement
Since metadata lock is supported, we can remove the support of the amending transaction.

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #40381

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
remove the support of the amending transaction
```

### CODE REVIEW

File: domain/schema_validator.go [hunk 1]
```diff
@@ -164,19 +164,18 @@ func (s *schemaValidator) Update(leaseGrantTS uint64, oldVer, currVer int64, cha
 // isRelatedTablesChanged returns the result whether relatedTableIDs is changed
 // from usedVer to the latest schema version.
 // NOTE, this function should be called under lock!
-func (s *schemaValidator) isRelatedTablesChanged(currVer int64, tableIDs []int64) (transaction.RelatedSchemaChange, bool) {
-	res := transaction.RelatedSchemaChange{}
+func (s *schemaValidator) isRelatedTablesChanged(currVer int64, tableIDs []int64) bool {
```
Comment (cfzjywxk) [inline domain/schema_validator.go:167]: Maybe we could add some comments to demonstrate that the schema validator now is only used when MDL is turned off and MDL would be enabled by default.

**Review by cfzjywxk — APPROVED** [review]
Please reference the corresponding issue in the description.

**Review by zyguan — APPROVED** [review]

File: domain/schema_validator.go (hunk 1 above)
Comment (wjhuang2016) [reply domain/schema_validator.go:167]: We should refine the comment later

File: domain/schema_validator.go [hunk 2]
```diff
@@ -198,22 +197,15 @@ func (s *schemaValidator) isRelatedTablesChanged(currVer int64, tableIDs []int64
 	}
 	if len(changedTblMap) > 0 {
 		tblIds := make([]int64, 0, len(changedTblMap))
-		actionTypes := make([]uint64, 0, len(changedTblMap))
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply domain/schema_validator.go:201]: Okey, we can refine it later.

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**